core/api/service/api_service.cpp

```diff
@@ -81,81 +157,51 @@ namespace kagome::api {
-                        else
-                          self->logger_->error("process Json data failed => {}",
-                                               response.error().message());
-                      });
+                  sendEvent(self->server_,
+                            session,
+                            self->logger_,
+                            set_id,
+                            kRpcEventSubscribeStorage,
+                            self->createStateStorageEvent(key, data, block));
                 }
               });
 
-          session_context.events_subscription->setCallback(
+          session_context->events_subscription->setCallback(
               [wp](uint32_t set_id,
                    SessionPtr &session,
                    const auto &key,
                    const auto &header) {
                 if (auto self = wp.lock()) {
-                  jsonrpc::Value::Struct p;
-                  p["result"] = api::makeValue(header);
-                  p["subscription"] = api::makeValue(set_id);
-
-                  jsonrpc::Request::Parameters params;
-                  params.push_back(std::move(p));
-                  if (key == primitives::SubscriptionEventType::kNewHeads) {
-                    self->server_->processJsonData(
-                        "chain_newHead", params, [&](const auto &response) {
-                          if (response.has_value())
-                            session->respond(response.value());
-                          else
-                            self->logger_->error(
-                                "process Json data failed => {}",
-                                response.error().message());
-                        });
-                  }
-                  if (key
-                      == primitives::SubscriptionEventType::kFinalizedHeads) {
-                    self->server_->processJsonData(
-                        "chain_finalizedHead",
-                        params,
-                        [&](const auto &response) {
-                          if (response.has_value())
-                            session->respond(response.value());
-                          else
-                            self->logger_->error(
-                                "process Json data failed => {}",
-                                response.error().message());
-                        });
+                  std::string_view name;
+                  switch (key) {
+                    case primitives::SubscriptionEventType::kNewHeads: {
+                      name = kRpcEventNewHeads;
+                    } break;
+                    case primitives::SubscriptionEventType::kFinalizedHeads: {
+                      name = kRpcEventFinalizedHeads;
+                    } break;
+                    case primitives::SubscriptionEventType::kRuntimeVersion: {
+                      name = kRpcEventRuntimeVersion;
+                    } break;
+                    default:
```

Comment:
dangerous. `default` should be unreachable, right?

Reply:
Assert checks this case.